### PR TITLE
Kdesktop 638 tmp file remain in tmp after download [Linux]

### PR DESCRIPTION
**Pull request by herve-er**

### CODE REVIEW

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 1]
```diff
@@ -110,15 +111,15 @@ bool DownloadJob::canRun() {
...
         _exitCode = ExitCode::SystemError;
         _exitCause = ExitCause::FileAccessError;
         return false;
     }
 
     if (_isCreate && exists) {
-        LOGW_DEBUG(_logger, L"Item: " << Utility::formatSyncPath(_localpath).c_str()
-                                      << L" already exist. Aborting current sync and restart.");
+        LOGW_DEBUG(_logger,
+                   L"Item: " << Utility::formatSyncPath(_localpath) << L" already exist. Aborting current sync and restart.");
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:122]:
```suggestion
                   L"Item with " << Utility::formatSyncPath(_localpath) << L" already exists. Aborting current sync and restarting.");
```

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 2]
```diff
@@ -209,7 +210,7 @@ bool DownloadJob::handleResponse(std::istream &is) {
         SyncPath tmpPath;
         IoError ioError = IoError::Success;
         if (!IoHelper::tempDirectoryPath(tmpPath, ioError)) {
-            LOGW_WARN(_logger, L"Failed to get temporary directory path: " << Utility::formatIoError(tmpPath, ioError).c_str());
+            LOGW_WARN(_logger, L"Failed to get temporary directory tmpPath: " << Utility::formatIoError(tmpPath, ioError));
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:213]: Why `tmpPath`?

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 3]
```diff
@@ -430,56 +431,56 @@ bool DownloadJob::createLink(const std::string &mimeType, const std::string &dat
...
         if (targetPath == _localpath) {
-            LOGW_DEBUG(_logger, L"Cannot create symlink on itself: " << Utility::formatSyncPath(_localpath).c_str());
+            LOGW_DEBUG(_logger, L"Cannot create symlink on itself: " << Utility::formatSyncPath(_localpath));
             return false;
         }
 
-        LOGW_DEBUG(_logger, L"Create symlink: " << Utility::formatSyncPath(targetPath).c_str() << L", "
-                                                << Utility::formatSyncPath(_localpath).c_str());
+        LOGW_DEBUG(_logger,
+                   L"Create symlink: " << Utility::formatSyncPath(targetPath) << L", " << Utility::formatSyncPath(_localpath));
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:439]:
```suggestion
                   L"Create symlink with target " << Utility::formatSyncPath(targetPath) << L", " << Utility::formatSyncPath(_localpath));
```

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 4]
```diff
@@ -528,10 +523,15 @@ bool DownloadJob::moveTmpFile(const SyncPath &path, bool &restartSync) {
...
             // In that case, try to copy the file instead.
             ec.clear();
-            std::filesystem::copy(path, _localpath, std::filesystem::copy_options::overwrite_existing, ec);
+            std::filesystem::copy(tmpPath, _localpath, std::filesystem::copy_options::overwrite_existing, ec);
+            bool removed = removeTmpFile(tmpPath);
             if (ec) {
-                LOGW_WARN(_logger, L"Failed to copy: " << Utility::formatSyncPath(_localpath).c_str() << L", "
-                                                       << Utility::formatStdError(ec).c_str());
+                LOGW_WARN(_logger,
+                          L"Failed to copy: " << Utility::formatSyncPath(_localpath) << L", " << Utility::formatStdError(ec));
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:530]:
```suggestion
                          L"Failed to copy to " << Utility::formatSyncPath(_localpath) << L", " << Utility::formatStdError(ec));
```

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 5]
```diff
@@ -528,10 +523,15 @@ bool DownloadJob::moveTmpFile(const SyncPath &path, bool &restartSync) {
...
+                return false;
+            }
+            if (!removed) {
+                LOGW_WARN(_logger, L"Failed to remove: " << Utility::formatSyncPath(tmpPath));
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:534]:
```suggestion
                LOGW_WARN(_logger, L"Failed to remove " << Utility::formatSyncPath(tmpPath));
```

File: src/libsyncengine/jobs/network/API_v2/downloadjob.cpp [hunk 6]
```diff
@@ -528,10 +523,15 @@ bool DownloadJob::moveTmpFile(const SyncPath &path, bool &restartSync) {
             // The sync might be on a different file system than tmp folder.
             // In that case, try to copy the file instead.
             ec.clear();
-            std::filesystem::copy(path, _localpath, std::filesystem::copy_options::overwrite_existing, ec);
+            std::filesystem::copy(tmpPath, _localpath, std::filesystem::copy_options::overwrite_existing, ec);
+            bool removed = removeTmpFile(tmpPath);
```
[the remark this answers is not in the document]
Comment (herve-er) [reply src/libsyncengine/jobs/network/API_v2/downloadjob.cpp:531]: Done here: [8c57f1f](https://github.com/Infomaniak/desktop-kDrive/pull/389/commits/8c57f1fba60295ba98112808ae8d6d8c864f776d)

**Review by ChristopheLarchier — APPROVED** [review]

**Comment by mrlem** [comment]
Still experiencing this issue with the 3.6.9 AppImage. kdrive* files keep filling my /tmp.

**Pull request merged**